.github/workflows/library_interop_tests.yml

```
@@ -0,0 +1,183 @@
+
+      - uses: actions/checkout@v3
+      # The specification submodule is private so we don't have access, but we don't need
+      # it to verify the Dafny code. Instead we manually pull the submodule we DO need.
+      - run: git submodule update --init libraries
+
+      # Set up runtimes
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Setup Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: 17
+
+      - name: Setup Dafny
+        uses: dafny-lang/setup-dafny-action@v1.6.1
+        with:
+          dafny-version: ${{ inputs.dafny }}
+
+      - name: test
+        run: |
+          languages="java net"
+          for i in $languages; do
+            echo $i
+          done
+
+      # Build implementation for each runtime
+      - name: Build ${{ matrix.library }} implementation in Java
+        shell: bash
+        working-directory: ./${{ matrix.library }}
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make build_java CORES=$CORES
+
+      - name: Build ${{ matrix.library }} implementation in .NET
+        shell: bash
+        working-directory: ./${{ matrix.library }}
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make transpile_net
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.2
+
+      # Test Vectors need to call KMS
+      - name: Configure AWS Credentials for Tests
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-MPL-Dafny-Role-us-west-2
+          role-session-name: InterOpTests
+
+      - name: Create Manifests
+        working-directory: ./${{ matrix.library }}
+        run: make test_generate_vectors_${{ matrix.languages }}
+
+      - name: Create Encrypt Manifests
+        working-directory: ./${{ matrix.library }}
+        run: make test_encrypt_vectors_${{ matrix.languages }}
+
+      - name: Upload Encrypt Manifest and keys.json files
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{matrix.os}}_vector_artifact_${{matrix.languages}}
+          path: ./${{matrix.library}}/runtimes/${{matrix.languages}}/*.json
+
+  testInteroperablity:
+    needs: generateEncryptVectors
+    strategy:
+      matrix:
+        library: [TestVectorsAwsCryptographicMaterialProviders]
+        os: [
+            # TODO just test on mac and ubuntu for now
+            # windows-latest,
+            ubuntu-latest,
+            macos-latest,
+          ]
+        languages: [java, net]
```

```suggestion
        encrypting_language: [java, net]
        decrypting_language: [java, net]
```
I think we need this to actually do interop, right now GHA is only decrypting NET with NET and Java with Java